cogs/economy.py

```diff
@@ -0,0 +1,79 @@
+import discord
+import time
+from discord.ext import commands
+from datetime import datetime
+import math
+
+class Economy(commands.Cog):
+    def __init__(self, bot):
+        self.bot = bot
+        self.storage = self.bot.get_cog('Storage')
+        self.voice_logs = self.storage.get_voice_log_cache()
+        self.voice_logs = {}
+        
+    # Listener that updates when a user joins and leaves a voice channel
+    @commands.Cog.listener()
+    async def on_voice_state_update(self, member, before, after):
+        # First time a user ever joins a voice channel in the server
+        if member.id not in self.voice_logs:
+            self.voice_logs[member.id] = {'member_obj': member,'start_time': datetime.utcnow(), 'total_time': 0, 'cactus_points': 0}
```

Comment:
Q: this also gets called when a member leaves a channel for the first time, should this be only done when someone enters a channel for the first time?